computer_vision/flip_augmentation.py

```diff
@@ -0,0 +1,117 @@
+import random
+import cv2
+import os
+import glob
+from typing import List
+
+"""
+Flip image and bounding box for computer vision task
+https://paperswithcode.com/method/randomhorizontalflip
+"""
+
+# Params
+LABEL_DIR = ''
+IMAGE_DIR = ''
+OUTPUT_DIR = ''
+FLIP_TYPE = 1  # (0 is vertical, 1 is horizontal)
+
+
+def main() -> None:
+    """
+    Get images list and annotations list from input dir.
+    Update new images and annotations.
+    Save images and annotations in output dir.
+    """
+    img_paths, annos = get_dataset(LABEL_DIR, IMAGE_DIR)
+    print('Processing...')
+    new_image, new_annos, path = update_image_and_anno(
+        img_paths, annos, FLIP_TYPE)
+
+    for index in range(len(new_image)):
+        # Get random string code: '7b7ad245cdff75241935e4dd860f3bad'
+        letter_code = random_chars(32)
+        file_name = path[index].split('/')[-1].rsplit('.', 1)[0]
+        cv2.imwrite(OUTPUT_DIR + f"/{file_name}_FLIP_{letter_code}.jpg", new_image[index], [cv2.IMWRITE_JPEG_QUALITY, 85])
+        print(f'Success {index+1}/{len(new_image)} with {file_name}')
+        annos_list = []
+        for anno in new_annos[index]:
+            obj = f'{anno[0]} {anno[1]} {anno[2]} {anno[3]} {anno[4]}'
+            annos_list.append(obj)
+        with open(OUTPUT_DIR + f"/{file_name}_FLIP{letter_code}.txt", "w") as outfile:
+            outfile.write("\n".join(line for line in annos_list))
+
+
+def get_dataset(label_dir: str, img_dir: str) -> List:
+    """
+    - label_dir <type: str>: Path to label include annotation of images
+    - img_dir <type: str>: Path to folder contain images
+    Return <type: list>: List of images path and labels 
+    """
+    img_paths = []
+    labels = []
+    for label_file in glob.glob(os.path.join(label_dir, '*.txt')):
+        label_name = label_file.split('/')[-1].rsplit('.', 1)[0]
+        f = open(label_file, 'r')
+        obj_lists = f.readlines()
```

Comment:
Opening files but not closing them is not a good idea so use `with open() as` to automate the closing of files.
```suggestion
        with open(label_file, 'r') as in_file:
            obj_lists = in_file.readlines()
```